### PR TITLE
Overload more operators in R for ndarray

### PR DESCRIPTION
**Pull request by hetong007**

### CODE REVIEW

File: R-package/R/ndarray.R [hunk 1]
```diff
@@ -1,12 +1,43 @@
...
   })
-  setMethod("+", signature(e1="Rcpp_MXNDArray", e2="Rcpp_MXNDArray"), function(e1, e2) {
+  setMethod("+", signature(e1 = "Rcpp_MXNDArray", e2 = "Rcpp_MXNDArray"), function(e1, e2) {
     mx.nd.internal.plus(e1, e2)
   })
+  setMethod("+", signature(e1 = "numeric", e2 = "Rcpp_MXNDArray"), function(e1, e2) {
+    mx.nd.internal.plus.scalar(e2, e1)
+  })
+  setMethod("-", signature(e1 = "Rcpp_MXNDArray", e2 = "numeric"), function(e1, e2) {
+    mx.nd.internal.minus.scalar(e1,e2)
```
Comment (tqchen) [inline R-package/R/ndarray.R]: be a bit consistent with style, space after comma

**Comment by tqchen** [comment]
cc @thirdwing

**Comment by thirdwing** [comment]
Both method is OK.

As I remembr, `+.Rcpp_MXDArray` is for S3 and S4, `setMethods` is only for S4.

**Comment by tqchen** [comment]
Rcpp_MXNDArray is  S4 object

**Comment by thirdwing** [comment]
That is why I said both are OK. We can follow this method.

PS: You won't see warnings below using `+.Rcpp_MXDArray`.

```
** R
** demo
** inst
** preparing package for lazy loading
in method for ‘+’ with signature ‘e1="Rcpp_MXNDArray",e2="numeric"’: no definition for class “Rcpp_MXNDArray”
in method for ‘+’ with signature ‘e1="Rcpp_MXNDArray",e2="Rcpp_MXNDArray"’: no definition for class “Rcpp_MXNDArray”
in method for ‘+’ with signature ‘e1="numeric",e2="Rcpp_MXNDArray"’: no definition for class “Rcpp_MXNDArray”
in method for ‘-’ with signature ‘e1="Rcpp_MXNDArray",e2="numeric"’: no definition for class “Rcpp_MXNDArray”
in method for ‘-’ with signature ‘e1="Rcpp_MXNDArray",e2="Rcpp_MXNDArray"’: no definition for class “Rcpp_MXNDArray”
in method for ‘-’ with signature ‘e1="numeric",e2="Rcpp_MXNDArray"’: no definition for class “Rcpp_MXNDArray”
in method for ‘*’ with signature ‘e1="Rcpp_MXNDArray",e2="numeric"’: no definition for class “Rcpp_MXNDArray”
in method for ‘*’ with signature ‘e1="Rcpp_MXNDArray",e2="Rcpp_MXNDArray"’: no definition for class “Rcpp_MXNDArray”
in method for ‘*’ with signature ‘e1="numeric",e2="Rcpp_MXNDArray"’: no definition for class “Rcpp_MXNDArray”
in method for ‘/’ with signature ‘e1="Rcpp_MXNDArray",e2="numeric"’: no definition for class “Rcpp_MXNDArray”
in method for ‘/’ with signature ‘e1="Rcpp_MXNDArray",e2="Rcpp_MXNDArray"’: no definition for class “Rcpp_MXNDArray”
in method for ‘/’ with signature ‘e1="numeric",e2="Rcpp_MXNDArray"’: no definition for class “Rcpp_MXNDArray”
```

**Comment by tqchen** [comment]
I thought the warning will not happen if the setmethods are in https://github.com/dmlc/mxnet/blob/master/R-package/R/ndarray.R#L4 ? It did not happen on my side

**Comment by tqchen** [comment]
Because init.nd.array is called in onLoad after the library loading

**Comment by thirdwing** [comment]
Defining a `init.ndarray.methods` and calling it during startup is not a "common" strategy, but it is OK.

I removed it and saw warnings.

**Comment by tqchen** [comment]
I see, this was really due to our tricky requirement of handling the loading orders.. I thought we can go for the S4 way as it is consider more advanced?

We can think of a more formal way later. But I am afraid we do need a init function solution, due to library loading order problem

**Pull request merged**